nailgun/entities.py

```diff
@@ -1096,6 +1135,23 @@ def __init__(self, server_config=None, **kwargs):
         }
         super(ContentView, self).__init__(server_config, **kwargs)
 
+    def read(self, entity=None, attrs=None, ignore=()):
+        """Fetch an attribute missing from the server's response.
+
+        For more information, see `Bugzilla #1237257
+        <https://bugzilla.redhat.com/show_bug.cgi?id=1237257>`_.
+
+        """
+        if (getattr(self._server_config, 'version', parse_version('6.1')) <
```

Comment:
Yeah. It definitely can be a mess. And the `nailgun.entities` module is huge and will continue to trend toward growth in the forseeable future. I can see a couple possible solution. One solution is to start making use of API specifications in the same way that python-foreman does. If nothing else, those specifications can be used to automatically populate fields. Another solution is to do something wild like make (all of? only some of?) the classes in `nailgun.entities` empty shells that just look at which version is being instantiated and delegate the instantiation process out to some other class. This isn't as outlandish as it seems:

```python
>>> from packaging.version import parse
>>> type(parse('1.0.0'))
<class 'packaging.version.Version'>
>>> type(parse('invalid version'))
<class 'packaging.version.LegacyVersion'>
```